File: backend/src/main/java/com/inventory/backend/repository/NotificationThresholdRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.NotificationThreshold;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface NotificationThresholdRepository extends JpaRepository<NotificationThreshold, Long> {
    List<NotificationThreshold> findByActiveTrueOrderByIdAsc();

    boolean existsByCartridgeModelIdAndDepartmentIdIsNull(Long cartridgeModelId);
    boolean existsByCartridgeModelIdAndDepartmentId(Long cartridgeModelId, Long departmentId);

    boolean existsByCartridgeModelIdAndDepartmentIdIsNullAndIdNot(Long cartridgeModelId, Long id);
    boolean existsByCartridgeModelIdAndDepartmentIdAndIdNot(Long cartridgeModelId, Long departmentId, Long id);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/NotificationThresholdRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.NotificationThreshold
import org.springframework.data.jpa.repository.JpaRepository

interface NotificationThresholdRepository : JpaRepository<NotificationThreshold, Long> {
    fun findByActiveTrueOrderByIdAsc(): List<NotificationThreshold>
    fun existsByCartridgeModelIdAndDepartmentIdIsNull(cartridgeModelId: Long): Boolean
    fun existsByCartridgeModelIdAndDepartmentId(cartridgeModelId: Long, departmentId: Long): Boolean
    fun existsByCartridgeModelIdAndDepartmentIdIsNullAndIdNot(cartridgeModelId: Long, id: Long): Boolean
    fun existsByCartridgeModelIdAndDepartmentIdAndIdNot(
        cartridgeModelId: Long,
        departmentId: Long,
        id: Long,
    ): Boolean
}
